tests/irdl/test_declarative_assembly_format.py

```diff
@@ -455,6 +455,68 @@ class OptionalPropertyOp(IRDLOperation):
     check_equivalence(program, generic_program, ctx)
 
 
+@pytest.mark.parametrize(
+    "program, generic_program",
+    [
+        (
+            "test.optional_unit_attr_prop",
+            '"test.optional_unit_attr_prop"() : () -> ()',
+        ),
+        (
+            "test.optional_unit_attr_prop unit_attr",
+            '"test.optional_unit_attr_prop"() <{"unit_attr"}> : () -> ()',
+        ),
+    ],
+)
+def test_optional_unit_attr_property(program: str, generic_program: str):
+    """Test the parsing of optional UnitAttr operands"""
+
+    @irdl_op_definition
+    class OptionalUnitAttrPropertyOp(IRDLOperation):
+        name = "test.optional_unit_attr_prop"
+        unit_attr = opt_prop_def(UnitAttr)
+
+        assembly_format = "(`unit_attr` $unit_attr^)? attr-dict"
+
+    ctx = MLContext()
+    ctx.load_op(OptionalUnitAttrPropertyOp)
+    ctx.load_dialect(Test)
+
+    check_roundtrip(program, ctx)
+    check_equivalence(program, generic_program, ctx)
+
+
+@pytest.mark.parametrize(
+    "program, generic_program",
+    [
+        (
+            "test.optional_unit_attr",
+            '"test.optional_unit_attr"() : () -> ()',
+        ),
+        (
+            "test.optional_unit_attr unit_attr",
+            '"test.optional_unit_attr"() <{"unit_attr"}> : () -> ()',
+        ),
+    ],
+)
+def test_optional_unit_attr_attribute(program: str, generic_program: str):
+    """Test the parsing of optional UnitAttr operands"""
+
+    @irdl_op_definition
+    class OptionalUnitAttrOp(IRDLOperation):
+        name = "test.optional_unit_attr"
+        unit_attr = opt_prop_def(UnitAttr)
+
+        assembly_format = "(`unit_attr` $unit_attr^)? attr-dict"
```

Comment:
Could you please make the strings different to disambiguate? Maybe something like this?

```suggestion
        unit_attr_prop = opt_prop_def(UnitAttr)

        assembly_format = "(`unit_attr_str` $unit_attr_prop^)? attr-dict"
```

Reply:
Before doing this, I'd note that this would diverge from how the other tests do it, as for the ones immediately above and below

Reply:
OK fair enough, please disregard